src/coreclr/pal/src/exception/remote-unwind.cpp

```diff
@@ -2213,6 +2213,57 @@ get_proc_name(unw_addr_space_t as, unw_word_t addr, char *bufp, size_t buf_len,
     return -UNW_ENOINFO;
 }
 
+#ifdef FEATURE_USE_SYSTEM_LIBUNWIND
+
+// Function typedef for unw_get_proc_info_in_range
+typedef int (*unw_get_proc_info_in_range_fn)(unw_word_t start_ip, unw_word_t end_ip, unw_word_t eh_frame_table, unw_word_t eh_frame_table_len, unw_word_t exidx_frame_table, unw_word_t exidx_frame_table_len, unw_addr_space_t as, unw_word_t ip, unw_proc_info_t *pi, int need_unwind_info, void *arg);
+
+// Static variables to cache the function pointer and the library handle
+static unw_get_proc_info_in_range_fn cached_unw_get_proc_info_in_range = NULL;
+static bool has_failed_to_load = false; // Track if loading has failed
+
+// Function to attempt to load unw_get_proc_info_in_range dynamically
+static bool
+TryGet_unw_get_proc_info_in_range()
+{
+    if (cached_unw_get_proc_info_in_range != NULL) return true;
+
+    // If we already failed to load before, do not retry
+    if (has_failed_to_load) return false;
+
+    void* handle = dlopen("libunwind.so.8", RTLD_LAZY);
```

Comment:
I wonder - instead of trying two libunwind SONAMEs, what do you think about using RTLD_DEFAULT pseudo handle in the dlsym instead?